aiaccel/hpo/algorithms/nelder_mead_algorithm.py

```diff
@@ -0,0 +1,201 @@
+    values: list[float]
+
+    def __init__(
+        self,
+        search_space: dict[str, tuple[float, float]],
+        coeff: NelderMeadCoefficient | None = None,
+        rng: np.random.RandomState | None = None,
+        block: bool = False,
+        timeout: int | None = None,
+    ) -> None:
+        self._search_space = search_space
+        self.coeff = coeff if coeff is not None else NelderMeadCoefficient()
+
+        self._rng = rng if rng is not None else np.random.RandomState()
+
+        self.generator = iter(self._generator())
+        self.lock = threading.Lock()
+
+        self.results: queue.Queue[tuple[npt.NDArray[np.float64], float, bool]] = queue.Queue()
+
+        self.block = block
+        self.timeout = timeout
+
+        self.simplex_size = len(self._search_space) + 1
+
+    def get_vertex(self) -> npt.NDArray[np.float64]:
+        with self.lock:
+            vertex = next(self.generator)
+
+        if vertex is None:
+            raise NelderMeadEmpty("Cannot generate new vertex now. Maybe get_vertex is called in parallel.")
+
+        return vertex
+
+    def put_value(
+        self,
+        vertex: npt.NDArray[np.float64],
+        value: float,
+        enqueue: bool = False,
+    ) -> None:
+        self.results.put((vertex, value, enqueue))
+
+    def _collect_enqueued_results(
+        self,
+        vertices: list[npt.NDArray[np.float64]] | None = None,
+        values: list[float] | None = None,
+    ) -> tuple[list[npt.NDArray[np.float64]], list[float]]:
+        vertices = [] if vertices is None else vertices
+        values = [] if values is None else values
+
+        while True:
+            try:
+                vertex, value, enqueue = self.results.get(block=False)
+                assert enqueue
+
+                vertices.append(vertex)
+                values.append(value)
+            except queue.Empty:
+                break
+
+        return vertices, values
+
+    def _wait_for_results(
+        self,
+        num_waiting: int,
+    ) -> Generator[None, None, list[float]]:
+        # collect results
+        vertices, values = list[npt.NDArray[np.float64]](), list[float]()
+        enqueued_vertices, enqueued_values = list[npt.NDArray[np.float64]](), list[float]()
+        while len(values) < num_waiting:
+            try:
+                vertex, value, enqueue = self.results.get(block=self.block)
+                if enqueue:
+                    enqueued_vertices.append(vertex)
+                    enqueued_values.append(value)
+                else:
+                    vertices.append(vertex)
+                    values.append(value)
+            except queue.Empty:
+                yield None
+
+        enqueued_vertices, enqueued_values = self._collect_enqueued_results(enqueued_vertices, enqueued_values)
+
+        # check if enqueued vertices change ordering
+        if len(self.values) == 0 or len(enqueued_values) > 0 and min(enqueued_values) < max(list(self.values)):
```

Comment:
self.valuesはlistにしたので，list(...) 不要ですね．
```suggestion
        if len(self.values) == 0 or len(enqueued_values) > 0 and min(enqueued_values) < max(self.values):
```